test/module/shared_model/builders/protobuf/transport_builder_test.cpp

```diff
@@ -250,3 +259,19 @@ TEST_F(TransportBuilderTest, ProposalCreationTest) {
       },
       [](const Error<std::string> &) { FAIL(); });
 }
+
+/**
+ * @given valid proto object of empty block
+ * @when transport builder constructs model object from it
+ * @then original and built objects are equal
+ */
+TEST_F(TransportBuilderTest, EmptyBlockCreationTest) {
+  auto orig_model = createEmptyBlock();
+  testTransport<decltype(orig_model), validation::DefaultEmptyBlockValidator>(
+      orig_model,
+      [&orig_model](const Value<decltype(orig_model)> &model) {
+        ASSERT_EQ(model.value.getTransport().SerializeAsString(),
+                  orig_model.getTransport().SerializeAsString());
+      },
+      [](const Error<std::string> &) { FAIL(); });
+}
```

Comment:
in the task "Both types have Block protobuf as backend, but empty one should check during stateless validation that it is empty and should not have transactions field"

I think the empty block with transaction test is needed

Reply:
Transport test included to demonstrate creation of EmptyBlock objects. 
Now there are no builders in shared_model for EmptyBlock, the only way to create EmptyBlock is to use TransportBuilder<BlockVariantType, AnyBlockValidator>, where BlockVariantType is variant<EmptyBlock, Block>. TransportBuilder checks if transactions' set is empty in protobuf object and creates either EmptyBlock or normal block